specifications/20200329-progressive-delivery.org

```diff
@@ -0,0 +1,99 @@
+#+TITLE: Progressive Delivery
+#+AUTHOR: Sergio Schvezov <sergio.schvezov@canonical.com>
+#+DATE: [2020-03-29]
+
+* Scope
+*Progressive Delivery* is scoped to defining how /progressive releases/ are
+executed and how their /status/ is displayed.
+
+* Enabling the feature
+As an experimental feature, an option must be used to enable this feature as
+defined on [[file:20200316-enabling-experimental-features.org][Enabling Experimental Features]].
+
+Such that a /progressive/ =status= can only be shown or a /progressive/
+=release= executed if combined with the experimental flag:
+
+#+BEGIN_SRC sh
+snapcraft <status|release> --experimental-progressive-delivery
+#+END_SRC
+
+Or with the following environment variable defined:
+#+BEGIN_SRC sh
+SNAPCRAFT_EXPERIMENTAL_PROGRESSIVE_DELIVERY=y snapcraft <status|release>
+#+END_SRC
+
+When the flag is used, the following shall be displayed using
+=snapcraft.cli.echo.warning=:
+
+#+BEGIN_SRC sh
+*EXPERIMENTAL* progressive releases in use.
+#+END_SRC
+
+* User Experience
+** Showing Progress
+Progress shall be displayed when used when the definitions in [[*Enabling the feature][Enabling the feature]]
+are satisfied.
+
+When run, for the case of a snap that has a /progressive release/ on the =edge=
+channel set to 80% with 71% for the /progressive release/ and 23% of the
+population on the actual release, the following behavior shall be observed:
+
+#+BEGIN_SRC sh
+$ snapcraft staging servers> snapcraft status candycane
+Track     Arch    Channel    Version    Revision    Progress
+latest    all     stable     -          -           -
+                  candidate  -          -           -
+                  beta       0.6        8           23 → 20%
+                             10         13          71 → 80%
+                  edge       ↑          ↑           -
+#+END_SRC
+
+If a /progressive release/ is done on =edge= while =edge= is currently
+following =beta=, then following information shall be observed:
```

Comment:
I think your examples are backwards.

Reply:
How so, this is exactly a verbatim copy of what we discussed in that meeting in Frankfurt

Reply:
oic

Reply:
when I did the oic I was looking at the table above, the example below is correct. Revision 11 is progressively released on edge, there is no current release on edge.